aci/resource_aci_fabricnodeidentp.go

```diff
@@ -242,19 +282,32 @@ func resourceAciFabricNodeMemberUpdate(ctx context.Context, d *schema.ResourceDa
 	if Role, ok := d.GetOk("role"); ok {
 		fabricNodeIdentPAttr.Role = Role.(string)
 	}
-	fabricNodeIdentP := models.NewFabricNodeMember(fmt.Sprintf("controller/nodeidentpol/nodep-%s", serial), "uni", desc, fabricNodeIdentPAttr)
-
-	fabricNodeIdentP.Status = "modified"
 
-	err := aciClient.Save(fabricNodeIdentP)
-
-	if err != nil {
-		return diag.FromErr(err)
+	if Commission, ok := d.GetOk("commission"); ok {
+		if Commission.(string) == "no" {
+			decommissionFabricNodeMemberFromController("false", "no", d, m)
+		} else {
+			attachedToSwitch, err := verifyNodeAttachedToSwitch(aciClient, serial)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			if attachedToSwitch {
+				decommissionFabricNodeMemberFromController("false", "yes", d, m)
+			} else {
+				fabricNodeIdentP := models.NewFabricNodeMember(fmt.Sprintf("controller/nodeidentpol/nodep-%s", serial), "uni", desc, fabricNodeIdentPAttr)
+				fabricNodeIdentP.Status = "modified"
+
+				err = aciClient.Save(fabricNodeIdentP)
+				if err != nil {
+					return diag.FromErr(err)
+				}
+
+				d.SetId(fabricNodeIdentP.DistinguishedName)
```

Comment:
Why is the following within the else block of attachedToSwitch? Wouldn't this affect update of other attributes if  Commission.(string) is no during update?
```
fabricNodeIdentP := models.NewFabricNodeMember(fmt.Sprintf("controller/nodeidentpol/nodep-%s", serial), "uni", desc, fabricNodeIdentPAttr)
				fabricNodeIdentP.Status = "modified"

				err = aciClient.Save(fabricNodeIdentP)
				if err != nil {
					return diag.FromErr(err)
				}

				d.SetId(fabricNodeIdentP.DistinguishedName)
```

Reply:
if commission is set to no -> then the node will be decommissioned from the switch, you can't update other parameters and set commission is is no in same task.

Reply:
Can you update if the node is attached to the switch? I'm only asking because I'm not sure of the behaviour..

Reply:
You can update the node if it is attached to a switch and is already commissioned,

if it is not commissioned, then it can't be updated